### ISSUE TITLE
On macOS pthreads is not found

### ISSUE THREAD

**Issue by Tieske** [luarocks/luarocks#1380]
### Please fill out the following:
- macOS Big Sur, 11.6.2
- luarocks: `/usr/local/Cellar/openresty@1.19.3.2/1.19.3.2/luarocks/bin/luarocks 3.5.0`

### Reproduce:
```
$ luarocks install lua-llthreads2 0.1.6-1
Installing https://luarocks.org/lua-llthreads2-0.1.6-1.src.rock

Error: Could not find library file for PTHREAD
  No file libpthread.a in /usr/local/lib
  No file libpthread.dylib in /usr/local/lib
  No file matching libpthread.so.* in /usr/local/lib
  No file libpthread.a in /usr/lib
  No file libpthread.dylib in /usr/lib
  No file matching libpthread.so.* in /usr/lib
  No file libpthread.a in /lib
  No file libpthread.dylib in /lib
  No file matching libpthread.so.* in /lib
You may have to install PTHREAD in your system and/or pass PTHREAD_DIR or PTHREAD_LIBDIR to the luarocks command.
Example: luarocks install lua-llthreads2 PTHREAD_DIR=/usr/local
$
```

### Workaround
```
ln -s /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib/libpthread.tbd /usr/local/lib/libpthread.dylib
```

Since macOS Big Sur, the system no longer has shared libraries as files in the file system. The OS has a dynamic cache of libraries. Luarocks searches for libraries by looking for actual files, so it fails (I think it should instead attempt a dlopen). Since all it needs is to find the file to pass that point, you can create a symlink from the library stub.

**Comment by hishamhm** [comment]
I believe a simpler alternative would be to add 
`"/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib"` to `external_deps_dirs` and `"lib?.tbd"` to `external_deps_patterns` to the default LuaRocks configuration in macOS?

**Comment by hishamhm** [comment]
> I think it should instead attempt a dlopen

dlopen takes a file name, so it's a chicken-and-egg problem.

> I believe a simpler alternative would be to add
"/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib" to external_deps_dirs and "lib?.tbd" to external_deps_patterns to the default LuaRocks configuration in macOS?

Could you give this a try and report if this works? Thanks!

**Comment by hishamhm** [comment]
This should be fixed in 3.9.0+, via commits a208f73967ad45c498c57e31ca27d29a48c10603, 17b51d22ab887ee76662ca7565239f214f1ade05 and fbd3566a312e647cde57b5d774533731e1aa844d. Let me know if the problem istill persists!
